utils/caution/caution.go

```diff
@@ -0,0 +1,26 @@
+// package caution provides utility functions for handling errors and closing
+// resources.
+// This package should be only be used for sitations where errors are unexpected
+// and should be ignored or panic
```

Comment:
This comment is not correct:

`CloseAndReportError` is meant to be used in production code for every error, to facilitate error management of the pattern `defer f.Close()`. `ExecuteAndReportError` is a generalization for types that do not implement the `Closer` interface.

The interface is designed to be used in this way:
Given a code
```
func F(....) error{
    [...]
    defer f.Close()
    [...]
}
```

The new functions can be easily introduced by naming the return variable and using the symbol name:
```
func F(....) (err error) {
    [...]
    defer CloseAndReportError(&err, f, "failed to close f")
    [...]
}
```

Reply:
documentation will be improved in next commit